src/cli.rs

```diff
@@ -64,6 +69,121 @@ fn check_path(
     })
 }
 
+/// This will generate a clap subcommand ([`App`]) that can be used for operations on CRDs.
+///
+/// Currently two arguments are supported:
+/// * `print`: This will print the schema to stdout
+/// * `save`: This will save the schema to a file
+///
+/// The resulting subcommand can be handled by the [`self::handle_crd_subcommand`] method.
+
+/// # Arguments
+///
+/// * `name`: Name of the CRD
+///
+/// returns: App
+///
+/// # Examples
+///
+/// ```
+/// use kube::CustomResource;
+/// use schemars::JsonSchema;
+/// use serde::{Serialize, Deserialize};
+///
+/// #[derive(Clone, CustomResource, Debug, JsonSchema, Serialize, Deserialize)]
+/// #[kube(
+///     group = "foo.stackable.tech",
+///     version = "v1",
+///     kind = "FooCluster",
+///     namespaced
+/// )]
+/// pub struct FooClusterSpec {
+///     pub name: String,
+/// }
+///
+/// let command = stackable_operator::cli::generate_crd_subcommand::<FooCluster>();
+/// ```
+pub fn generate_crd_subcommand<'a, 'b, T>() -> App<'a, 'b>
+where
+    T: CustomResourceExt,
+{
+    let kind = T::api_resource().kind;
+
+    SubCommand::with_name(&kind.to_lowercase())
```

Comment:
If I'm not misreading this a subcommand per custom resource would be generated? 

So from a user perspective, I'd need to run this once per crd (cluster, start, stop, restart, rebalance, ...) to print or save the crd?

```
spark-operator restart print
spark-operator stop save stop.yaml
```

I assume this is in order to nest these under a `crd` command from the operator code? 
Are we planning to add a `--all` parameter as well, that would probably need to be done by the operator itself?

Reply:
We could make a builder out of this and basically have a CrdSubcommandBuilder to which we could add new Subcommands which it could then put into a list and automatically add the `all` function.... I'll think about it for a bit.

Reply:
We can do that but we can't generalize the handling I believe

Reply:
> We can do that but we can't generalize the handling I believe

This is in reference to the `--all` flag I assume? 
If so, then I think I agree. I think we could introduce a new struct to represent CustomResources along these lines:

```
struct StackableCrd {
  definition: String,
  resource: ApiResource,
}
```
Then that could be exposed via a trait on the cluster object and retrieved in a `Vec<StackableCrd>` by the framework to handle the all parameter, or passed into the function doing the handling.

I hesitate to suggest this for just the cli handling though, but this might actually turn out to be usefull down the line as well...

I have something like this in the command handling code:

```
pub trait HasCommands {
    fn get_command_types() -> Vec<ApiResource>;
}
```
Which is half of what I suggested above already ...

Reply:
I'm not sure I understand.
`definition` is the schema?

The problem with your proposal (if I understand it correctly) is that we don't actually have an instance of an object. All we have is the type.

When generating the command we could generate and store the schema and then later don't use the types anymore but do everything based on strings (this sentence probably doesn't make much sense to you...)

I'm not sure if it's worth it. I'm in favor of merging as is. We can improve later.

Reply:
I totally agree, lets merge this as is!

Regarding my idea, I was thinking something along these lines as an associated function on the cluster level structs:

```rust
impl HasCommands for SparkCluster {
    fn get_command_types() -> Vec<StackableCrd> {
        vec![
            StackableCrd{ definition: Start::get_schema(), resource: Start::api_resource() },
            StackableCrd{ definition: Stop::get_schema(), resource: Stop::api_resource() },
            StackableCrd{ definition: Restart::get_schema(), resource: Restart::api_resource() },
        ]
    }
}
```

That would give us something to iterate over in the framework and should contain everything we need to create the crds.